src/main/scala/io/iohk/ethereum/network/AuthInitiateMessage.scala

```diff
@@ -0,0 +1,33 @@
+package io.iohk.ethereum.network
+
+import akka.util.ByteString
+import io.iohk.ethereum.crypto._
+import org.spongycastle.math.ec.ECPoint
+
+object AuthInitiateMessage {
+
+  def decode(input: Array[Byte]) = {
+    AuthInitiateMessage(
+      signature = ECDSASignature.decode(input.take(65)),
+      ephemeralPublicHash = ByteString(input.slice(65, 65 + 32)),
+      publicKey = curve.getCurve.decodePoint(Array(4.toByte) ++ input.slice(97, 97 + 64)),
+      nonce = ByteString(input.slice(161, 161 + 32)),
+      knownPeer = input(193) == 1)
+  }
+}
+
+case class AuthInitiateMessage(
+    signature: ECDSASignature,
+    ephemeralPublicHash: ByteString,
+    publicKey: ECPoint,
+    nonce: ByteString,
+    knownPeer: Boolean) {
+
+  def encode(): ByteString = {
```

Comment:
By convention methods with no side effects and no params should be declared without '()'. When you see a method with () and no params you are not free to replace it with a val as it may have side effects. In this case, there are none, so it should be replaced with a lazy val so that the effort of calculating it is never repeated unnecessarily.